src/parsers/manifest/dash/indexes/timeline/construct_timeline_from_previous_timeline.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import log from "../../../../../log";
+import { IIndexSegment } from "../../../utils/index_helpers";
+import constructTimelineFromElements from "./construct_timeline_from_elements";
+import convertElementToIndexSegment from "./convert_element_to_index_segment";
+import findFirstCommonStartTime from "./find_first_common_start_time";
+import parseSElement, {
+  IParsedS,
+} from "./parse_s_element";
+
+export default function constructTimelineFromPreviousTimeline(
+  newElements : HTMLCollection,
+  prevTimeline : IIndexSegment[],
+  scaledPeriodStart : number
+) : IIndexSegment[] {
+
+  // Find first index in both timeline where a common segment is found.
+  const commonStartInfo = findFirstCommonStartTime(prevTimeline, newElements);
+  if (commonStartInfo === null) {
+    log.warn("DASH: Cannot perform \"based\" update. Common segment not found.");
+    return constructTimelineFromElements(newElements, scaledPeriodStart);
+  }
+  const { prevSegmentsIdx,
+          newElementsIdx,
+          repeatNumberInPrevSegments,
+          repeatNumberInNewElements } = commonStartInfo;
+
+  /** Guess of the number of elements in common. */
+  const numberCommonEltGuess = prevTimeline.length - prevSegmentsIdx;
+  const lastCommonEltNewEltsIdx = numberCommonEltGuess + newElementsIdx - 1;
+  if (lastCommonEltNewEltsIdx >= newElements.length) {
+    log.info("DASH: Cannot perform \"based\" update. New timeline too short");
+    return constructTimelineFromElements(newElements, scaledPeriodStart);
+  }
+
+  // Remove elements which are not available anymore
```

Comment:
/** */ ?

Reply:
Same than for the other one. I thought that was more of a code comment than newTineline's documentation.